src/base/static/components/molecules/form-field-types/map-drawing-toolbar.js

```diff
@@ -35,26 +45,89 @@ const deleteToolLabels = {
   "create-polygon": "deletePolygonToolLabel",
 };
 
+// NOTE: The ColorPicker component, which consumes these two ToolbarButton
+// components, tries to attach a ref to its children, so these components
+// need to be classes.
+class StrokeColorpickerToolbarButton extends Component {
+  render() {
+    return (
+      <ToolbarButton
+        classes={classNames(
+          "map-drawing-toolbar__tool-icon",
+          "map-drawing-toolbar__tool-icon--colorpicker",
+          {
+            "map-drawing-toolbar__tool-icon--colorpicker--active":
+              this.props.activeColorpicker ===
+              constants.DRAW_STROKE_COLORPICKER_NAME,
+          },
+        )}
+        label={this.props.label}
+        icon="/static/css/images/colorpicker-icon.svg"
+        onClick={this.props.onClick}
+      />
+    );
+  }
+}
+
+StrokeColorpickerToolbarButton.propTypes = {
+  activeColorpicker: PropTypes.string,
+  label: PropTypes.string.isRequired,
+  onClick: PropTypes.func,
+};
+
+class FillColorpickerToolbarButton extends Component {
+  render() {
+    return (
+      <ToolbarButton
+        classes={classNames(
+          "map-drawing-toolbar__tool-icon",
+          "map-drawing-toolbar__tool-icon--colorpicker",
+          {
+            "map-drawing-toolbar__tool-icon--colorpicker--active":
+              this.props.activeColorpicker ===
+              constants.DRAW_FILL_COLORPICKER_NAME,
+          },
+        )}
+        label={this.props.label}
+        icon="/static/css/images/colorpicker-icon.svg"
+        onClick={this.props.onClick}
+      />
+    );
+  }
+}
+
+FillColorpickerToolbarButton.propTypes = {
+  activeColorpicker: PropTypes.string,
+  label: PropTypes.string.isRequired,
+  onClick: PropTypes.func,
+};
+
 class MapDrawingToolbar extends Component {
   componentDidMount() {
+    this.props.updateDrawModeActive(true);
     this.props.resetDrawingToolbarState();
     this.props.setMarkers(this.props.markers);
     this.drawUpdateListener = emitter.addListener(
```

Comment:
We still have one event emitter here, so the map drawing field can update its data in the same way all the other form fields do (i.e. via its `onChange` handler).